storage/mysql/tree_storage.go

```diff
@@ -34,7 +34,7 @@ import (
 
 // These statements are fixed
 const (
-	insertSubtreeMultiSQL = `INSERT INTO Subtree(TreeId, SubtreeId, Nodes, SubtreeRevision) ` + placeholderSQL
+	insertSubtreeMultiSQL = `REPLACE INTO Subtree(TreeId, SubtreeId, Nodes, SubtreeRevision) ` + placeholderSQL
```

Comment:
Should we use `INSERT INTO ... ON DUPLICATE KEY UPDATE Nodes = VALUES(Nodes)` instead?
On my benchmarks I didn't see any difference, but I think it's [safer](https://code.openark.org/blog/mysql/replace-into-think-twice/comment-page-5#comments) to use it. Wdyt?

Reply:
Thanks. I've switched over to this version. I have a minor concern that the `VALUES()` syntax is unfavourable in both MySQL[1] and to a lesser degree MariaDB[2] in the newest versions, but for the time being they both support it so let's roll with it.

Tested this with the following docker images:
  - mysql:5
  - mysql:8.0
  - mariadb:10.5

[1] https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html - "The use of VALUES() to refer to the new row and columns is deprecated beginning with MySQL 8.0.20, and is subject to removal in a future version of MySQL"
[2] https://mariadb.com/kb/en/values-value/ - In MariaDB 10.3.3 this function was renamed to VALUE(), because it's incompatible with the standard Table Value Constructors syntax ... The VALUES() function can still be used even from MariaDB 10.3.3 but only in INSERT ... ON DUPLICATE KEY UPDATE statements; it's a syntax error otherwise.